rustworkx-core/src/generators/hexagonal_lattice_graph.rs

```diff
@@ -88,95 +93,88 @@ where
     if rows == 0 || cols == 0 {
         return Ok(G::with_capacity(0, 0));
     }
-    let mut rowlen = rows;
-    let mut collen = cols;
 
-    // Needs two times the number of nodes vertically
-    rowlen = 2 * rowlen + 2;
-    collen += 1;
-    let num_nodes = rowlen * collen - 2;
+    if periodic && (cols % 2 == 1 || rows < 2 || cols < 2) {
+        return Err(InvalidInputError {});
+    }
+
+    // Number of nodes in each vertical chain
+    let rowlen = if periodic {
+        2 * rows
+    } else {
+        // Note: in the non-periodic case the first and
+        // last vertical chains have (2 * rows + 1) nodes
+        2 * rows + 2
+    };
+
+    // Number of vertical chains
+    let collen = if periodic { cols } else { cols + 1 };
+
+    let num_nodes = if periodic {
+        rowlen * collen
+    } else {
+        rowlen * collen - 2
+    };
```

Comment:
You can rewrite all the variables with a single match on that condition using destructuring e.g. https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=ef7020b44cc19135583d87a1dcf6d1b2

Reply:
Forgot to mention that I did this as well in 6497d44.